apps/fxc-front/src/app/logic/track.ts

```diff
@@ -46,3 +58,94 @@ async function fetchAndReturnGroupIds(url: string, options?: RequestInit): Promi
   }
   return Array.from(groupIds);
 }
+
+export function scoreTrack(track: RuntimeTrack){
+	const request: ScoreRequest = {track: track, league: currentLeague(store.getState())}
+	getScoreWorker(store.dispatch).postMessage(request)
+}
+
+let scoreWorker: Worker | undefined;
+
+function getScoreWorker(dispatch: AppDispatch): Worker {
+  if (!scoreWorker){
+    scoreWorker = new ScoreWorker() as Worker;
+    scoreWorker.onmessage = (msg: MessageEvent<ScoreResponse & RuntimeTrackId>) => {
+      const data = msg.data;
+      const {encodedRoute, indexes} = getEncodeRouteAndIndices(data.scoreInfo);
+      const score: Score = {
```

Comment:
disclaimer: I haven't checked the details and I might be wrong - I haven't touch this part of the code in a long time.

What I was thinking is that the worker would only return the points/indicies.

Given the points the planner should return the same score.

Reply:
this part have been modify. Tell me if the new version is more convenient.